Src/IronPython/Modules/Builtin.cs

```diff
@@ -1401,26 +1401,23 @@ public static object eval(CodeContext/*!*/ context, [NotNull]FunctionCode code,
                 return DynamicHelpers.GetPythonTypeFromType(typeof(ReversedEnumerator));
             }
         }
-        
-        public static double round(double number) {
-            return MathUtils.RoundAwayFromZero(number);
-        }
-
-        public static double round(double number, int ndigits) {
-            return PythonOps.CheckMath(number, MathUtils.RoundAwayFromZero(number, ndigits));
-        }
 
-        public static double round(double number, BigInteger ndigits) {
-            int n;
-            if (ndigits.AsInt32(out n)) {
-                return round(number, n);
+        public static object round(CodeContext/*!*/ context, object number) {
+            if (!PythonOps.TryGetBoundAttr(context, number, "__round__", out var func)) {
```

Comment:
Maybe consider adding fast paths for the common case. For example:
```C#
if (number is double d) return DoubleOps.__round__(d);
```
This will likely give a significant perf improvement.

Reply:
Done.